doc/source/cluster-index.rst

```diff
@@ -1,15 +1,105 @@
-Deploying Ray
-=============
+Multi-node Ray (Overview)
```

Comment:
first things first, we want to structure this so that all files related to the autoscaler are under `/cluster`. For instance:
-  `doc/source/cluster/index.rst`
-  `doc/source/cluster/configuration.rst`
-  `doc/source/cluster/cloud.rst`
-  `doc/source/cluster/cluster-managers.rst`

With the structure, it makes it very easy for people to navigate the tree and know exactly where something should go. By removing from top level you also make contributions easier ("just throw it in the cluster folder").

Reply:
Index should just be what this file is.